Getting Started with Form (#1485)
DevExpress/devextreme-documentation: concepts/05 Widgets/Form/01 Getting Started with Form/20 Populate the Form Fields.md

> The Form generates a [simple item](/Documentation/ApiReference/UI_Widgets/dxForm/Item_Types/SimpleItem/) for each field in the [formData](/Documentation/ApiReference/UI_Widgets/dxForm/Configuration/#formData) object. A simple item is a label-editor pair. 

Reviewer: It seems that the same information is given in the paragraph below. Shouldn't this paragraph be removed?

> To populate form fields, assign an object to the **formData** property. Each field in this object will be visualized as a label-editor pair. Such pairs are called "simple items". The Form automatically defines which editors it will generate for fields in the **formData** object. The editor's type depends on the data type of the field's value. The Form in the following example will create the [TextBox](/Documentation/ApiReference/UI_Widgets/dxTextBox/), [NumberBox](/Documentation/ApiReference/UI_Widgets/dxNumberBox/), and [DateBox](/Documentation/ApiReference/UI_Widgets/dxDateBox/) editors:

Reviewer: In your version, you go from form fields to label-editor pairs to simple items then back to editors. I suppose this structure stems from one of my earlier suggestions. However, it doesn't work as we need to end the paragraph focusing on the editors. It would be better to mention simple items earlier than talking about label-editor pairs and then editors alone.

I also took the liberty of describing default editors in more detail. Here's what I came up with:

```suggestion
To populate form fields, assign an object to the [formData](/Documentation/ApiReference/UI_Widgets/dxForm/Configuration/#formData) property. Each data field in this object produces a [simple item](/Documentation/ApiReference/UI_Widgets/dxForm/Item_Types/SimpleItem/)&mdash;a label-editor pair in which the label displays the data field name and the editor edits the data field value.

The Form chooses default editors based on the field's data type: [TextBox](/Documentation/ApiReference/UI_Widgets/dxTextBox/) for string values, [NumberBox](/Documentation/ApiReference/UI_Widgets/dxNumberBox/) for numbers, [DateBox](/Documentation/ApiReference/UI_Widgets/dxDateBox/) for dates. You can use the [editorType](/Documentation/ApiReference/UI_Widgets/dxForm/Item_Types/SimpleItem/#editorType) property to specify an editor explicitly. In the following example, the Form uses default editors:
```

        bootstrap: [AppComponent]
    })
    export class AppModule { }

##### Vue

    <!-- tab: App.vue -->
    <template>
        <DxForm 
            :form-data="employee">
        </DxForm>
    </template>

    <script>
    import 'devextreme/dist/css/dx.common.css';
    import 'devextreme/dist/css/dx.light.css';

    import { DxForm } from 'devextreme-vue/form';

    const employee = {
        name: 'John Heart',
        officeNumber: 901,
        hireDate: new Date(2012, 4, 13)
    };

    export default {
        components: {
            DxForm
        },
        data: {
            return: {
                employee
            }
        }
    }
    </script>

##### React

    <!-- tab: App.js -->
    import React from 'react';
    import 'devextreme/dist/css/dx.common.css';
    import 'devextreme/dist/css/dx.light.css';

    import {
        Form
    } from 'devextreme-react/form';

    const employee = {
        name: 'John Heart',
        officeNumber: 901,
        hireDate: new Date(2012, 4, 13)
    };

    const App = () => {
        return (
            <Form
                formData={employee}>
            </Form>
        );
    }

    export default App;

> ---

Reviewer: Add a couple of sentences about the result that the reader should see if he/she did everything correctly.

> The **Form** generates a [simple item]() for each field in the [formData]() object. A simple item is a label-editor pair. The **Form** automatically defines the editor's type depending on the data type of the **formData** object's field. The **Form** in the following example will create the [TextBox](), [NumberBox](), and [DateBox]() editors:

Reviewer: This is too descriptive. Use imperative mood when you want the reader to do something. Tell how to populate form fields straightaway:

> To populate form fields, assign an object to the [formData] property. 

Then you need to go from form fields to simple items because this isn't obvious.

> Each field in this object will be visualized as a label-editor pair. Such pairs are called "simple items".

---

> The **Form** automatically defines the editor's type depending on the data type of the **formData** object's field.

Try to break this long sentence in two.

---
##### jQuery

    <!-- tab: index.js -->
    $(function() {
        $("#form").dxForm({
            formData: {
                name: "John Heart",

>                 officeNumber: "901",

Reviewer: Are you sure that this will generate a NumberBox?

Author: Will fix it

                hireDate: new Date(2012, 4, 13)
            }
        });
    });

##### Angular

    <!-- tab: app.component.html -->
    <dx-form

>         [(formData)]="employee"

Reviewer: Must this be two-way binding or one-way will do?

>     >

Reviewer: ```suggestion
        [(formData)]="employee">
```

    })
    export class AppComponent {
        employee = {
            name: "John Heart",
            officeNumber: "901",
            hireDate: new Date(2012, 4, 13)
        }
    }

    <!-- tab: app.module.ts -->
    import { BrowserModule } from '@angular/platform-browser';
    import { NgModule } from '@angular/core';
    import { AppComponent } from './app.component';

    import { DxFormModule } from 'devextreme-angular';

    @NgModule({
        declarations: [
            AppComponent
        ],
        imports: [
            BrowserModule,
            DxFormModule
        ],
        providers: [ ],
        bootstrap: [AppComponent]
    })
    export class AppModule { }

##### Vue

    <!-- tab: App.vue -->
    <template>
        <div id="app-container">
            <DxForm 
                :form-data="employee">
            </DxForm>
        </div>
    </template>

    <script>
    import 'devextreme/dist/css/dx.common.css';
    import 'devextreme/dist/css/dx.light.css';

    import { DxForm } from 'devextreme-vue/form';

    export default {
        components: {
            DxForm
        },
        data: {
            return: {

>                 employee = {

Reviewer: ```suggestion
                employee: {
```

                    name: "John Heart",
                    officeNumber: "901",
                    hireDate: new Date(2012, 4, 13)
                }
            }
        }
    }
    </script>

##### React

    <!-- tab: App.js -->
    import React from 'react';
    import 'devextreme/dist/css/dx.common.css';
    import 'devextreme/dist/css/dx.light.css';

    import {
        Form
    } from 'devextreme-react/form';

>     const employee = {

Reviewer: I guess this should be a variable since its fields are going to be edited via the Form.

To populate form fields, assign an object to the [formData](/Documentation/ApiReference/UI_Widgets/dxForm/Configuration/#formData) property. Each data field in this object produces a [simple item](/Documentation/ApiReference/UI_Widgets/dxForm/Item_Types/SimpleItem/)&mdash;a label-editor pair in which the label displays the data field name and the editor edits the data field value.

> The Form chooses default editors based on the field's data type: [TextBox](/Documentation/ApiReference/UI_Widgets/dxTextBox/) for string values, [NumberBox](/Documentation/ApiReference/UI_Widgets/dxNumberBox/) for numbers, [DateBox](/Documentation/ApiReference/UI_Widgets/dxDateBox/) for dates. You can use the [editorType](/Documentation/ApiReference/UI_Widgets/dxForm/Item_Types/SimpleItem/#editorType) property to specify an editor explicitly. In the following example, the Form uses default editors:

Reviewer 2: Note: "chooses" depends on a data field **value**. dxForm will create TextBoxes if you pass an object with 'null/undefined' values in properties (or no properties). A am not sure that we should mention it at this stage but have to add a note about it.

Author: In my opinion, Getting Started is not the right place to note about such occasions. I think that if we don't have such a note in our docs, it's better to create a docissue.

@RomanTsukanov , what do you think?

---
##### jQuery

    <!-- tab: index.js -->
    $(function() {
        $("#form").dxForm({
            formData: {
                name: "John Heart",
                officeNumber: 901,
                hireDate: new Date(2012, 4, 13)
            }
        });
    });

##### Angular

    <!-- tab: app.component.html -->
    <dx-form
        [formData]="employee">
    </dx-form>

    <!-- tab: app.component.ts -->
    import { Component } from '@angular/core';

    @Component({
        selector: 'app-root',
        templateUrl: './app.component.html',
        styleUrls: ['./app.component.css']
    })
    export class AppComponent {
        employee = {

>             name: "John Heart",

Reviewer 3: single quote. There are the same mistake in the following samples.

            officeNumber: 901,
            hireDate: new Date(2012, 4, 13)
        }
    }

    <!-- tab: app.module.ts -->
    import { BrowserModule } from '@angular/platform-browser';
    import { NgModule } from '@angular/core';
    import { AppComponent } from './app.component';

    import { DxFormModule } from 'devextreme-angular';

    @NgModule({
        declarations: [
            AppComponent
        ],
        imports: [
            BrowserModule,
            DxFormModule
        ],
        providers: [ ],
        bootstrap: [AppComponent]
    })
    export class AppModule { }

##### Vue

    <!-- tab: App.vue -->
    <template>
        <div id="app-container">
            <DxForm 
                :form-data="employee">
            </DxForm>
        </div>
    </template>

    <script>
    import 'devextreme/dist/css/dx.common.css';
    import 'devextreme/dist/css/dx.light.css';

    import { DxForm } from 'devextreme-vue/form';

>     let employee = {

Reviewer 3: const?

Author: If we modify the form data at runtime, won't this object change?
